site/blog/tutorials/optimize-s3-endpoints.mdx

```diff
@@ -11,40 +11,45 @@ Are you using S3 buckets as part of your cloud deployments? How are you accessin
 
-When the VPC endpoints are enabled you can access your S3 buckets using this endpoint. In this post, we will walk you through control your buckets from an internal network with the desired security, and without the extra costs of a
-NAT gateway using a VPC endpoint and IaSQL. IaSQL is an [open-source](https://github.com/iasql/iasql) software tool that creates a two-way connection between an unmodified PostgreSQL database and an AWS account so you can manage your infrastructure from a database.
+However, creating <a href="https://docs.aws.amazon.com/vpc/latest/privatelink/vpc-endpoints-s3.html" target="_blank">gateway or interface VPC endpoints</a> for each region where your buckets are exposed is a more optimal solution.
+
+<img src="/img/optimize-s3-endpoints/s3_privatelink.png" style={{ maxWidth: 600 }} />
+
+When the VPC endpoints are enabled you can access your S3 buckets using this endpoint. In this post, we will walk you through control your buckets from an internal network with the desired security, and without the extra costs of a NAT gateway using a VPC endpoint and IaSQL. IaSQL is an [open-source](https://github.com/iasql/iasql) software tool that creates a two-way connection between an unmodified PostgreSQL database and an AWS account so you can manage your infrastructure from a database.
 
 <!-- truncate -->
 
 ### Why use VPC Interface Endpoints
 
-AWS VPC Interface Endpoints are a component of AWS's <a href="https://aws.amazon.com/privatelink">PrivateLink</a> infrastructure. AWS PrivateLink provides private connectivity between virtual private clouds (VPCs),
-supported AWS services, and your on-premises networks without exposing your traffic to the public internet.
+AWS VPC Interface Endpoints are a component of AWS's <a href="https://aws.amazon.com/privatelink">PrivateLink</a> infrastructure. AWS PrivateLink provides private connectivity between virtual private clouds (VPCs), supported AWS services, and your on-premises networks without exposing your traffic to the public internet.
+
+Relying on PrivateLink offers a set of advantages in terms of **cost**, **security** and **performance**.
+
+**Cost**
+
+When using NAT gateways, you are paying for the NAT gateway itself, and the bandwidth used to access the internet. A NAT gateway is a generic way for private instances to reach internet, and can be useful for generic usage, such
+as pulling dependencies, updating repositories, etc... But when used for reaching AWS resources directly, there are more optimal and cheaper solutions such as VPC endpoints.
+When using VPC endpoints, you are paying for the bandwidth used to access the internet, but not for the NAT gateway itself. Costs for NAT gateway depend on each region, but it should be over _0.045 USD/hour_ per GB processed + _0.045 USD/hour_ for the instance itself. Costs for VPC endpoints are charged at _0.01 USD/hour_ per GB processed, being cheaper after the first PB.
```

Comment:
> When using VPC endpoints, you are paying for the bandwidth used to access the internet, but not for the NAT gateway itself.

This sentence is confusing. I thought VPC endpoints were for accessing other VPCs and AWS services, not the general internet?

Reply:
no, i am trying to justify the more expensive costs of NAT gateway. I want to say that NAT gateway is expensive because it is more general purpose and can be used to access internet from private instances. While vpc endpoints can only be used to access public AWS resources, they are more limited, so they are cheaper